### PR TITLE
Maxpool-8 shell

### PR DESCRIPTION
**Pull request by tomdol**
### Details:
 - MaxPool-8 operator shell implementation
 - Common members and shape inference utilities moved to a base class that both MaxPool version derive from
 - Includes attribute visitor and type_prop tests
 - The implementation is based on https://github.com/openvinotoolkit/openvino/pull/5359 which might get adjusted in a few days

### Tickets:
 - 54658

### CODE REVIEW

**Comment by andrei-kochin** [comment]
@ilyachur @a-sidorova  Please review nGraph shell implementation for MaxPool-8

File: ngraph/core/include/ngraph/op/max_pool.hpp [hunk 1]
```diff
@@ -76,21 +56,85 @@ namespace ngraph
...
+                ///             as a starting point for calculation of the upper bound of allowed
+                ///             values of the indices output.
+                MaxPool(const Output<Node>& arg,
+                        const Strides& strides,
+                        const Strides& dilations,
+                        const Shape& pads_begin,
+                        const Shape& pads_end,
+                        const Shape& kernel,
+                        const op::RoundingType rounding_type = op::RoundingType::FLOOR,
+                        const PadType& auto_pad = op::PadType::EXPLICIT,
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/max_pool.hpp:99]: Why do we have a mix of passing enumerations by reference and by value?

File: ngraph/core/include/ngraph/op/max_pool.hpp [hunk 2]
```diff
@@ -76,21 +56,85 @@ namespace ngraph
...
+                        const element::Type& index_element_type = element::i64,
+                        const int64_t axis = 0,
+                        const float pads_value = -std::numeric_limits<float>::infinity());
+
+                bool visit_attributes(AttributeVisitor& visitor) override;
+                void validate_and_infer_types() override;
+
+                virtual std::shared_ptr<Node>
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/max_pool.hpp:107]: nitpick: virtual is redundant

File: ngraph/core/include/ngraph/op/util/max_pool_base.hpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                /// \param rounding_mode Whether to use ceiling or floor rounding type while
+                /// computing output shape.
+                /// \param auto_pad The pad type for automatically computing padding sizes.
+                MaxPoolBase(const Output<Node>& arg,
+                            const Strides& strides,
+                            const Shape& pads_begin,
+                            const Shape& pads_end,
+                            const Shape& kernel,
+                            op::RoundingType rounding_mode = op::RoundingType::FLOOR,
+                            const PadType& auto_pad = op::PadType::EXPLICIT);
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/util/max_pool_base.hpp:36]: The same mix

File: ngraph/core/src/op/max_pool.cpp [hunk 4]
```diff
@@ -279,3 +169,78 @@ bool op::v1::MaxPool::has_evaluate() const
     }
     return false;
 }
+
+// ------------------------------ V8 ------------------------------
+
+NGRAPH_RTTI_DEFINITION(op::v8::MaxPool, "MaxPool", 8);
```
Comment (ilyachur) [inline ngraph/core/src/op/max_pool.cpp:175]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v8::MaxPool, "MaxPool", 8, MaxPoolBase);
```

File: ngraph/core/src/op/util/max_pool_base.cpp [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                                  arg_shape.rank().get_max_length() - 2,
+                              "Expected pads_begin size to be equal to input size - 2. Got: ",
+                              m_pads_begin.size());
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<int64_t>(m_kernel.size()) ==
+                                  arg_shape.rank().get_max_length() - 2,
+                              "Expected kernel size to be equal to input size - 2. Got: ",
+                              m_kernel.size());
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<int64_t>(m_pads_end.size()) ==
```
Comment (a-sidorova) [inline ngraph/core/src/op/util/max_pool_base.cpp:80]:
```suggestion
                              static_cast<int64_t>(m_strides.size()) ==
```

Comment (tomdol) [reply ngraph/core/src/op/util/max_pool_base.cpp:80]: Good catch, thanks

File: ngraph/core/include/ngraph/op/max_pool.hpp (hunk 1 above)
Comment (tomdol) [reply ngraph/core/include/ngraph/op/max_pool.hpp]: Right, there were inconsistencies and I just copy and pasted them. Fixed.

**Review by a-sidorova — APPROVED** [review]

File: ngraph/core/src/op/max_pool.cpp [hunk 6]
```diff
@@ -14,44 +14,16 @@
...
-                                   m_auto_pad,
-                                   pads_end,
-                                   pads_begin);
-        new_pads_end = Shape(pads_end.begin(), pads_end.end());
-        new_pads_begin = Shape(pads_begin.begin(), pads_begin.end());
-    }
-    return update_auto_padding_succeed;
-}
-
 NGRAPH_RTTI_DEFINITION(op::v1::MaxPool, "MaxPool", 1);
```
Comment (ilyachur) [inline ngraph/core/src/op/max_pool.cpp:17]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v1::MaxPool, "MaxPool", 1,MaxPoolBase);
```

**Review by ilyachur — APPROVED** [review]

**Pull request merged**